### PR TITLE
Add tutorials about notifications (ref #353)

### PR DESCRIPTION
**Pull request by leplatrem**
early feedback or help would be greatly appreciated please ! 

/ping @n1k0 @Natim @almet @zwhitchcox :)

The best way is probably to checkout the branch and build the docs locally:

```
$ git checkout 353-notifications-documentation
$ make docs
$ open docs/_build/html/index.html
```

### CODE REVIEW

File: docs/tutorials/notifications-websockets.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+Now that *Kinto* runs locally and is configured to send events to *Pusher*, you
+should be able to see them in the *Debug Console* of your *Pusher dashboard*.
+
+For example, create an arbitrary record with `cURL <https://en.wikipedia.org/wiki/CURL>`_
+
+::
+
+    curl -i --user alice:secret -H "Content-Type: application/json" \
+         -X POST -d '{"data":{"name":"bob"}}' \
+         http://localhost:8888/v1/buckets/default/collections/tasks/records
```
Comment (leplatrem) [inline docs/tutorials/notifications-websockets.rst]: should probably be replaced by httpie like elsewhere

File: docs/overview.rst [hunk 2]
```diff
@@ -151,6 +151,21 @@ of objects.
...
 
+.. _overview-notifications:
+
+Notifications
+=============
+
+*Kinto* can execute some code when a particular event occurs.
+For example, when a record is created or updated in a particular collection.
+
+It can send a notifications to clients using WebSockets or fill a queue of messages
```
Comment (leplatrem) [inline docs/overview.rst]: typo: singular notification

File: docs/tutorials/notifications-websockets.rst [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+      console.log("Records deleted", data);
+    });
+
+
+Demos
+-----
+
+We've made several demos with live sync.
+
+For example, open these one in several browser windows and observe the
```
Comment (leplatrem) [inline docs/tutorials/notifications-websockets.rst]: ~~one~~

File: docs/overview.rst [hunk 4]
```diff
@@ -151,6 +151,21 @@ of objects.
...
+in Redis or execute any custom code of your choice, like for sending emails or pinging a
+third-party.
```
Comment (n1k0) [inline docs/overview.rst]: Nit: Adding links to *WebSockets* and *Redis* might help newcomers.

File: docs/tutorials/notifications-custom.rst [hunk 5]
```diff
@@ -0,0 +1,148 @@
+.. _tutorial-notifications-custom-code:
+
+How to run custom code on notifications ?
+=========================================
+
+Kinto is able to execute some custom code when a particular event occurs.
+For example, when a record is created or updated in a particular collection.
+
+This tutorial presents the basic steps to run some code:
```
Comment (n1k0) [inline docs/tutorials/notifications-custom.rst]: Nit: s/some//

File: docs/tutorials/notifications-custom.rst [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+
+* synchronously in Python;
+* asynchronously using a Redis queue, consumed via the language of your choice.
+
+
+Run synchronous code
+--------------------
+
+In this example, we will send an email to some administrator every time
```
Comment (n1k0) [inline docs/tutorials/notifications-custom.rst]: Nit: s/some/an

File: docs/tutorials/notifications-custom.rst [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        from = settings[prefix + 'from']
+        recipients = aslist(settings[prefix + 'recipients'])
+
+        return Listener(server, from, recipients)
+
+
+Add it to Python path
+'''''''''''''''''''''
+
+TBD
```
Comment (n1k0) [inline docs/tutorials/notifications-custom.rst]: Might be worth filling this section before landing ;)

File: docs/tutorials/notifications-custom.rst [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+    cliquet.event_listeners.send_email.server = user:password@mail.server.com
+    cliquet.event_listeners.send_email.from = postmaster@localhost
+    cliquet.event_listeners.send_email.recipients = kinto@yopmail.com
+
+
+
+Run asynchronous code
+---------------------
+
+TBD
```
Comment (n1k0) [inline docs/tutorials/notifications-custom.rst]: Same remark.

File: docs/tutorials/notifications-websockets.rst [hunk 9]
```diff
@@ -0,0 +1,119 @@
+.. _tutorial-notifications-websockets:
+
+How to setup push notifications using WebSockets ?
+==================================================
+
+This tutorial presents the basic steps to to implement live synchronisation of
```
Comment (n1k0) [inline docs/tutorials/notifications-websockets.rst]: s/to to/to

File: docs/tutorials/notifications-websockets.rst (hunk 1 above)
Comment (n1k0) [inline docs/tutorials/notifications-websockets.rst]: +1

**Comment by n1k0** [comment]
Looks good!

**Comment by leplatrem** [comment]
Code now works :) Ready to review/merge!

**Pull request merged**